ledger/simulation/trace.go

```diff
@@ -70,23 +70,36 @@ type ResultEvalOverrides struct {
 	AllowEmptySignatures bool
 	MaxLogCalls          *uint64
 	MaxLogSize           *uint64
+	ExtraAppBudget       *uint64
```

Comment:
minor: I think you could drop the pointer here too, since `ExtraAppBudget` tells you the additional budget and 0 is enough to indicate no extra budget was added